### PR TITLE
fstab-generator: Options= applies specifier expansion

### PR DESCRIPTION
**Pull request by poettering**
Let's document that this is the case, and properly escape % when we
generate Options= in the generator.

Fixes: #5086

### CODE REVIEW

File: man/systemd.mount.xml [hunk 1]
```diff
@@ -366,9 +366,9 @@
       <varlistentry>
         <term><varname>Options=</varname></term>
 
-        <listitem><para>Mount options to use when mounting. This takes
-        a comma-separated list of options. This setting is
-        optional.</para></listitem>
+        <listitem><para>Mount options to use when mounting. This takes a comma-separated list of options. This setting
+        is optional. Note that the usual specifier expansion is applied to this setting, literal percent characters
+        should hence be written as <literal>%%</literal>.</para></listitem>
```
Comment (evverx) [inline man/systemd.mount.xml:369]: https://github.com/systemd/systemd/pull/4835/commits/d107589cd2767c95fb1c794d4f90832301a297f5#diff-815691af24aeb3dba59fb6414f2ec484L357

Shouldn't we escape (and document) `What` too?

**Comment by poettering** [comment]
pushed an updated version, also adding What= escaping, as suggested by @evverx.

**Comment by evverx** [comment]
I'm reading https://github.com/systemd/systemd/pull/4835
https://github.com/systemd/systemd/pull/4835/commits/744bb5b1bea4d04363f7894e86701efdd75b8acb affects `RequiresMountsFor`.
Shouldn't we escape `RequiresMountsFor` in `fstab-generator`, `cryptsetup-generator` and so on?

**Comment by evverx** [comment]
Actually, I don't understand how does `x-systemd.*` work with escaping/unescaping

**Comment by poettering** [comment]
sure we can escape that stuff too, but let's maybe do that in a separate PR.

**Comment by poettering** [comment]
I posted a new bug tracking those remaining cases in #5290, but let's get this PR merged for now, as it at least fixes the most visible cases.

**Pull request merged**